src/__tests__/components/elements/text/text.spec.tsx

```diff
@@ -105,4 +121,35 @@ describe(UI_TYPE, () => {
 		expect(screen.getByText("This is a sanitized string")).toBeInTheDocument();
 		expect(consoleSpy).not.toBeCalled();
 	});
+
+	it("should be able to render view more button", async () => {
+		const childrenContent = ["apple", "berry", "cherry", "orange"];
+
+		const clientHeight = Object.getOwnPropertyDescriptor(HTMLElement.prototype, "clientHeight");
+		const scrollHeight = Object.getOwnPropertyDescriptor(HTMLElement.prototype, "scrollHeight");
+
+		Object.defineProperty(HTMLElement.prototype, "clientHeight", { configurable: true, value: 400 });
+		Object.defineProperty(HTMLElement.prototype, "scrollHeight", { configurable: true, value: 500 });
+		renderComponent({ children: childrenContent, maxLines: 3 });
+
+		const viewMoreButton = screen.queryByText("View more");
+		expect(viewMoreButton).toBeInTheDocument();
+	});
```

Comment:
understand that it's not possible to get the original heights, in that case, after the test completes, let's restore the heights back to zero

```
Object.defineProperty(HTMLElement.prototype, "clientHeight", { value: 0 });
Object.defineProperty(HTMLElement.prototype, "scrollHeight", { value: 0 });
```